### ISSUE TITLE
add support for verifying the response header is absent

### ISSUE THREAD

**Issue by prabirshrestha** [trillium-rs/trillium#145]
Is it possible to support `None` for a non existence header? Passing empty string will not work since the `None != Some("")`.

I added the compression middleware and wanted to do a sanity test.

```rust
#[test]
fn should_compress_html() {
    let handler = app();
    assert_headers!(
        get("/")
            .with_request_header(KnownHeaderName::AcceptEncoding, "gzip, deflate, br")
            .on(&handler),
        "content-encoding" => "br"
    );

    assert_headers!(
        get("/")
            .with_request_header(KnownHeaderName::AcceptEncoding, "gzip")
            .on(&handler),
        "content-encoding" => "gzip"
    );

    assert_headers!(
        get("/")
            .with_request_header(KnownHeaderName::AcceptEncoding, "")
            .on(&handler),
        "content-encoding" => None
    );

}

```

**Comment by jbr** [comment]
This is already implemented on main, will be released with the next minor revision of trillium-testing

**Comment by jbr** [comment]
closed with [trillium-testing@v0.4.0](https://github.com/trillium-rs/trillium/releases/tag/trillium-testing-v0.4.0)
